src/Graphql/Type/SchemaBuilder.php

```diff
@@ -294,16 +294,18 @@ private function convertType(Type $type, bool $input = false, string $mutationNa
      */
     private function getResourceObjectType(string $resourceClass, ResourceMetadata $resourceMetadata, bool $input = false, string $mutationName = null): GraphQLType
     {
-        $shortName = $resourceMetadata->getShortName();
-        if ($input) {
-            $shortName .= 'Input';
+        if (isset($this->graphqlTypes[$resourceClass][$mutationName][$input])) {
+            return $this->graphqlTypes[$resourceClass][$mutationName][$input];
         }
+
+        $shortName = $resourceMetadata->getShortName();
         if (null !== $mutationName) {
-            $shortName .= ucfirst($mutationName).'Mutation';
+            $shortName = $mutationName.ucfirst($shortName);
         }
-
-        if (isset($this->graphqlTypes[$resourceClass][$mutationName][$input])) {
-            return $this->graphqlTypes[$resourceClass][$mutationName][$input];
+        if ($input) {
+            $shortName .= 'Input';
```

Comment:
What if `$mutationName` is null but `$input` is true? You will concatenate a string on a non-existing variable. But could this situation happen?

Reply:
`$shortName` always exists.